config/auth.go

```diff
@@ -293,3 +300,64 @@ func (o *AuthOIDCConfig) Validate() error {
 	}
 	return o.HTTPClientConfiguration.Validate()
 }
+
+// AuthKerberosAuthzConfig is the authorization configuration for Kerberos
+// authentication
+type AuthKerberosAuthzConfig struct {
+	HTTPClientConfiguration `json:",inline" yaml:",inline"`
+	// Controls whether the authorization flow is enabled. If set to false
+	// all authenticated users are allowed in the service.
+	Enable bool `json:"enable" yaml:"enable" default:"false"`
+}
+
+func (k *AuthKerberosAuthzConfig) Validate() error {
+	if k.Enable {
+		return k.HTTPClientConfiguration.Validate()
+	}
+	return nil
+}
+
+// AuthKerberosClientConfig is the configuration for the Kerberos authentication method.
+type AuthKerberosClientConfig struct {
+	// Keytab is the path to the kerberos keytab
+	Keytab string `json:"keytab" yaml:"keytab" default:"/etc/krb5.keytab"`
+	// Acceptor is the name of the keytab entry to authenticate against.
+	// The value of this field needs to be in the form of `service/name`.
+	//
+	// The special value of `host` will authenticate clients only against
+	// the service `host/hostname` where hostname is the system hostname
+	// The special value of 'any' will authenticate against all keytab
+	// entries regardless of name
+	Acceptor string `json:"acceptor" yaml:"acceptor" default:"any"`
+	// EnforceUsername specifies whether to check that the username of the
+	// authenticated user matches the SSH username entered. If set to false
+	// the authorization server must be responsible for ensuring proper
+	// access control.
+	// **WARNING: If authorization is unset and this is set to false all
+	// authenticated users can log in to any account!**
+	EnforceUsername bool `json:"enforceUsername" yaml:"enforceUsername" default:"true"`
+	// CredentialCachePath is the path in which the kerberos credentials
+	// will be written inside the user containers. For kerberos to work
+	// correctly this should match the contents of the KRB5CC environment
+	// variable.
+	CredentialCachePath string `json:"credentialCachePath" yaml:"credentialCachePath" default:"/tmp/krb5cc"`
+	// AllowPassword controls whether kerberos-based password
+	// authentication should be allowed. If set to false only GSSAPI
+	// authentication will be permitted
+	AllowPassword bool `json:"allowPassword" yaml:"allowPassword" default:"true"`
+	// ConfigPath is the path of the kerberos configuration file. This is
+	// only used for password authentication.
+	ConfigPath string `json:"configPath" yaml:"configPath" default:"/etc/containerssh/krb5.conf"`
```

Comment:
KRB config is quite complex, I'd prefer not to at least until we have a compelling reason to make the change.
Additionally, this allows using the distribution/packaged config directly instead of translating it to containerssh cofig, which may be useful in some cases.